components/docker-auth/cmd/authn/authentication.go

```diff
@@ -105,15 +133,130 @@ func main() {
 	tokenValidity, err := validateToken(token, certificateInUse)
 	if err != nil {
 		glog.Errorf("Token is not valid - %s", err)
-		glogFlushAndExit(extension.ErrorExitCode)
+		extension.LogErrorAndExit(extension.ErrorExitCode)
 	}
 	glog.Info("signature verified")
 
 	if tokenValidity {
 		glog.Info("user successfully authenticated")
-		glogFlushAndExit(extension.SuccessExitCode)
+		extension.LogErrorAndExit(extension.SuccessExitCode)
 	} else {
 		glog.Error("authentication failed")
-		glogFlushAndExit(extension.ErrorExitCode)
+		extension.LogErrorAndExit(extension.ErrorExitCode)
+	}
+}
+
+// isJWT checks whether the token is jwt token or access token.
+func isJWT(conf extension.Conf) bool {
+	isJWTEnv := os.Getenv("IS_JWT")
+	var isJWT bool
+	if len(isJWTEnv) == 0 {
+		isJWT = conf.AuthConf.IsJWT
+	} else {
+		if isJWTEnv == "true" {
+			isJWT = true
+		} else if isJWTEnv == "false" {
+			isJWT = false
+		} else {
+			glog.Error("Wrong environment value given. The value should be either true or false")
+			extension.LogErrorAndExit(extension.ErrorExitCode)
+		}
+	}
+	return isJWT
+}
+
+// validateAccessToken is used to introspect the access token
+func validateAccessToken(token string, providedUsername string, conf extension.Conf) bool {
+	isHost, isPort := resolveISHostAndPort(conf)
+	url := "https://" + isHost + ":" + isPort + "/oauth2/introspect"
+	payload := strings.NewReader("token=" + token + "")
+	req, err := http.NewRequest("POST", url, payload)
+	if err != nil {
+		glog.Error("Error creating new request to the introspection endpoint: ", err)
+		extension.LogErrorAndExit(extension.ErrorExitCode)
+	}
+
+	username, password := resolveCredentials(conf)
+	req.SetBasicAuth(username, password)
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
I don't think this should be committed to our repos. This is only something we would do in a dev environment. Let's remove the line.